Dependency update

daniel-noland opened the pull request:
I bumped the dependencies to current everywhere I could find something out of date.

I didn't notice any regressions in this patch series although some minor refactoring was required.

I left the series as distinct, easily reverted patches in case something goes wrong.  Let me know if you want me to squash them.

jasonwhite commented on docker-compose.yml:
```diff
@@ -2,10 +2,10 @@
-    #build:
-    #  context: .
-    #  dockerfile: Dockerfile
+    image: rudolfs:local
+    build:
+     context: .
+     dockerfile: Dockerfile
```
Did you intend to commit these changes? Looks like this was just changed for local development.

daniel-noland replied:
Oh, nope!

That is me being sloppy with version control.

I'll fix it.

daniel-noland replied:
this should be sorted now

jasonwhite reviewed (changes requested):
Awesome! Thanks for this. Just one minor question before I merge it.

jasonwhite commented on src/storage/cached.rs:
```diff
@@ -29,7 +29,7 @@ use futures::{
     stream::{StreamExt, TryStreamExt},
 };
-use humansize::{file_size_opts as file_size, FileSize};
+use humansize;
```
This line can be removed.

daniel-noland replied:
true.  Will fix it

daniel-noland replied:
fixed

jasonwhite commented:
FYI, I fixed the clippy lint errors in the main branch.

daniel-noland commented:
> FYI, I fixed the clippy lint errors in the main branch.

I rebased onto `master` so clippy warnings should not bounce me in CI this time.

Thanks for sorting them :)

daniel-noland commented:
one note vs prior version of this PR:

I didn't even see the `faulty` feature when I did the prior batch of updates.

As a result my changes introduced several regressions I didn't catch on that code base.

I cleaned them up in the second to final commit in this series.  That commit is worth review at your convenience.

If you want me to move that to a different patch series that's fine.

Alternatively I can re-structure the patch series to make sure `faulty` doesn't regress as we walk the updates.  As this series stands there are commits in the tree where `faulty` won't build
